apps/meteor/tests/end-to-end/api/livechat/14-units.ts

```diff
@@ -409,4 +413,428 @@ import { IS_EE } from '../../../e2e/config/constants';
+					departmentUnit: { _id: unitId },
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.property('department').that.is.an('object');
+					expect(res.body.department).to.have.property('name', departmentName);
+					expect(res.body.department).to.have.property('type', 'd');
+					expect(res.body.department).to.have.property('_id');
+					departmentId = res.body.department._id;
+				});
+		};
+
+		before(async () => {
+			monitor1 = await createUser();
+			monitor2 = await createUser();
+			await createMonitor(monitor1.username);
+			monitor1Credentials = await login(monitor1.username, password);
+			await createMonitor(monitor2.username);
+			monitor2Credentials = await login(monitor2.username, password);
+			unit = await createUnit(monitor1._id, monitor1.username, []);
+		});
+
+		after(async () => Promise.all([deleteUser(monitor1), deleteUser(monitor2), deleteUnit(unit)]));
+
+		afterEach(() => {
+			if (departmentId) {
+				return deleteDepartment(departmentId);
+			}
+		});
+
+		it('should fail creating department when providing an invalid property in the department unit object', () => {
+			return request
+				.post(api('livechat/department'))
+				.set(credentials)
+				.send({
+					department: { name: 'Fail-Test', enabled: true, showOnOfflineForm: true, showOnRegistration: true, email: 'bla@bla' },
+					departmentUnit: { invalidProperty: true },
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('errorType', 'invalid-params');
+				});
+		});
+
+		it('should fail creating a department into an existing unit that a monitor does not supervise', async () => {
+			const departmentName = 'Test-Department-Unit2';
+
+			await createDepartmentInUnit(monitor2Credentials, departmentName, unit._id);
+			await testDepartmentsInUnit(unit._id, unit.name, 0);
+			await testDepartmentAncestors(departmentId, departmentName);
+		});
+
+		it('should succesfully create a department into an existing unit as a livechat manager', async () => {
+			const departmentName = 'Test-Department-Unit';
+
+			await createDepartmentInUnit(credentials, departmentName, unit._id);
+			await testDepartmentsInUnit(unit._id, unit.name, 1);
+			await testDepartmentAncestors(departmentId, departmentName, unit._id);
+		});
+
+		it('should succesfully create a department into an existing unit that a monitor supervises', async () => {
+			const departmentName = 'Test-Department-Unit3';
+
+			await createDepartmentInUnit(monitor1Credentials, departmentName, unit._id);
+
+			// Deleting a department currently does not decrease its unit's counter. We must adjust this check when this is fixed
```

Comment:
Can you please open a task for it?

Reply:
Sure, done! [Here](https://rocketchat.atlassian.net/browse/CORE-541) it is